modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/processor/ServiceRequestDataSampleProcessor.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.monitoring.internal.processor;
+
+import com.liferay.portal.kernel.monitoring.DataSampleProcessor;
+import com.liferay.portal.kernel.monitoring.MethodSignature;
+import com.liferay.portal.monitoring.internal.statistics.service.ServiceRequestDataSample;
+import com.liferay.portal.monitoring.internal.statistics.service.ServiceStatistics;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Renan Vasconcelos
+ */
+@Component(
+	enabled = false, property = "namespace=com.liferay.monitoring.Service",
+	service = ServiceRequestDataSampleProcessor.class
+)
+public class ServiceRequestDataSampleProcessor
+	implements DataSampleProcessor<ServiceRequestDataSample> {
+
+	@Override
+	public void processDataSample(
+		ServiceRequestDataSample serviceRequestDataSample) {
+
+		MethodSignature methodSignature =
+			serviceRequestDataSample.getMethodSignature();
+
+		String className = methodSignature.getClassName();
+
+		ServiceStatistics serviceStatistics = _serviceStatistics.get(className);
+
+		if (serviceStatistics == null) {
+			serviceStatistics = new ServiceStatistics(className);
+
+			_serviceStatistics.put(className, serviceStatistics);
+		}
+
+		serviceStatistics.processDataSample(serviceRequestDataSample);
+	}
+
+	private final Map<String, ServiceStatistics> _serviceStatistics =
+		new ConcurrentHashMap<>();
+
```

Comment:
Hey @RenanUmV, this seems wrong. We have the same map in the other class. Could you please check that? Thx